### ISSUE TITLE
Enabling IP Prefix doesn't work as expected.

### ISSUE THREAD

**Issue by uthark** [aws/amazon-vpc-cni-k8s#1836]
**What happened**:

```
{"level":"error","ts":"2022-02-01T00:15:06.311Z","caller":"ipamd/ipamd.go:1001","msg":"Failed to allocate a private IP/Prefix addresses on ENI eni-0d735700aade4ab83: InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request.\n\tstatus code: 400, request id: fd5f3b54-8d8f-4f9d-9df6-3de916f965c3"}
{"level":"error","ts":"2022-02-01T00:15:06.311Z","caller":"ipamd/ipamd.go:653","msg":"failed to allocate one IPv4 prefix on ENI eni-0d735700aade4ab83, err: InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request.\n\tstatus code: 400, request id: fd5f3b54-8d8f-4f9d-9df6-3de916f965c3: InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request.\n\tstatus code: 400, request id: fd5f3b54-8d8f-4f9d-9df6-3de916f965c3"}
{"level":"error","ts":"2022-02-01T00:15:06.311Z","caller":"ipamd/ipamd.go:653","msg":"Unable to attach IPs/Prefixes for the ENI, subnet doesn't seem to have enough IPs/Prefixes. Consider using new subnet or carve a reserved range using create-subnet-cidr-reservation"}
```

We are trying to enable the use of IP Prefixes.
What we did.
* Created IP Prefixes CIDR reservations in the subnet.
* Configured AWS plugin to use CIDR reservation for new nodes (i.e. node reports if IP Prefixes should be used and CNI plugin uses it during bootstrap).
* Subnet size is /19 (this is 512 /28 blocks). I created 51 reservations (10% of the subnet size).
* But when node boots, CNI fails with 
```
 InsufficientCidrBlocks: The specified subnet does not have enough free cidr blocks to satisfy the request
```

parameters from Cloudtrail:

```
    "requestParameters": {
        "networkInterfaceId": "eni-0d735700aade4ab83",
        "privateIpAddressesSet": {},
        "ipv4Prefixes": {},
        "ipv4PrefixCount": 1
    },
```

**What you expected to happen**:
CNI uses created CIDR reservations.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.21.8
- CNI Version - v1.10.1
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04.
- Kernel (e.g. `uname -a`): 5.4.0-1063-aws

**Comment by achevuru** [comment]
@uthark  Subnet CIDR reservations will succeed even when there are some IPs within those reserved subnets that are already in use. However, we will not be able to assign those prefixes to an ENI until all the IPs within those reserved subnets are free. Please check if the entire reserved subnet range is free.

```
You can create a prefix reservation even if some of the space in the reservation range is currently taken up secondary IP addresses. Once IP addresses from that space are released, EC2 won’t reassign them.
```
https://aws.amazon.com/blogs/containers/amazon-vpc-cni-increases-pods-per-node-limits/

**Comment by uthark** [comment]
yes, I read this and what got me confused is that at this point reservation is actually not available for use (and there is no API call to check).

We wrote helper script that would cross-reference secondary private ip addresses attached to ENI against IP reservation to detect if IP reservation is available for use.

I'd suggest to update AWS docs to indicate this caveat. (based on the docs I got an impression that AWS can manage routing properly in case single IP address with reservation is being used by other ENI.

closing the issue.
